write/write.c

```diff
@@ -0,0 +1,212 @@
+/*
+ * write.c - write to another user
+ *
+ * Copyright (C) 2024: Luiz Antônio Rangel (takusuman)
+ *
+ * SPDX-Licence-Identifier: Zlib
+ *
+ *	from Unix 32V /usr/src/cmd/write.c
+ *	November 6th, 1978.
+ *
+ * Copyright(C) Caldera International Inc. 2001-2002. All rights reserved.
+ *
+ * SPDX-Licence-Identifier: Caldera
+ */
+
+#include <limits.h>
+#include <pwd.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strmenta.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+#include <utmp.h>
+#include <utmpx.h>
+#define MAXNAMLEN _POSIX_LOGIN_NAME_MAX
+
+int	signum[] = {SIGHUP, SIGINT, SIGQUIT, 0};
+FILE	*tf;
+
+void	main(int argc, char *argv[]);
+void	eof(void);
+void	timout(void);
+int	ex(char* bp);
+int	sigs(void (*sig)(int));
+void	usage(void);
+
+void main(int argc, char *argv[]) {
```

Comment:
The code of this function seems to be obfuscated. Both in terms of lack of whitespace and randomly generated variable names.

Reply:
Which function you're talking about? ``sigs()``?
That's some damn witchcraft, I don't understand how it works and it has to be compiled as being C89 to work.
As I said before, I didn't made anything besides porting and, in the case of write.c, writing error messages to the standard error output (as Solaris, HP-UX and other modern implementations derived from UNIX v7 are doing).

Reply:
Oh, ``main()``? Well, UNIX v7 (or 7th, just a simple formality anyways) code is like that.
Anyway, I don't think there's nothing explicitly erred there, besides I think it isn't the best way to write code nowadays. I mean, "i" and "c" are vulgar iterators, "tfd" stands for "terminal file descriptor", "logcnt" for "login count" and "places" for the "number places where the user is logged in". The only problem I see is that there's almost no blank lines separating the code lines in blocks, but this is just historical remain and maybe it's not a big deal.

Reply:
"him"

Reply:
Yeah, the receiver of the message.